packages/shared-ux/chrome/navigation/types.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/**
+ * TODO move definition from the Share plugin to a package
+ * @private
+ */
+export interface SerializedLocator {
+  id: string;
+  params: object;
```

Comment:
I know it's early, but you could use a pair of generics for this and state.  Ignore me if there's more work to be done here.

Reply:
++
this seems like a good time to do it now